cpg_pipes/targets.py

```diff
@@ -0,0 +1,640 @@
+        """
+        return self.storage_provider.get_web_base(
+            dataset=self.stack, namespace=self.namespace, **kwargs
+        )
+
+    def get_web_url(self, **kwargs) -> str | None:
+        """
+        Get web base URL.
+        """
+        return self.storage_provider.get_web_url(
+            dataset=self.stack, namespace=self.namespace, **kwargs
+        )
+
+    def add_sample(
+        self,
+        id: str,  # pylint: disable=redefined-builtin
+        external_id: str | None = None,
+        participant_id: str | None = None,
+        sequencing_type: SequencingType | None = None,
+        meta: dict | None = None,
+        sex: Optional['Sex'] = None,
+        pedigree: Optional['PedigreeInfo'] = None,
+        alignment_input: AlignmentInput | None = None,
+    ) -> 'Sample':
+        """
+        Create a new sample and add it to the dataset.
+        """
+        if id in self._sample_by_id:
+            logger.debug(f'Sample {id} already exists in the dataset {self.name}')
+            return self._sample_by_id[id]
+
+        s = Sample(
+            id=id,
+            dataset=self,
+            external_id=external_id,
+            sequencing_type=sequencing_type,
+            participant_id=participant_id,
+            meta=meta,
+            sex=sex,
+            pedigree=pedigree,
+            alignment_input=alignment_input,
+        )
+        self._sample_by_id[id] = s
+        return s
+
+    def get_samples(self, only_active: bool = True) -> list['Sample']:
+        """
+        Get dataset's samples. Include only "active" samples, unless only_active=False
+        """
+        return [
+            s for sid, s in self._sample_by_id.items() if (s.active or not only_active)
+        ]
+
+    def get_job_attrs(self) -> dict[str, str]:
+        """
+        Attributes for Hail Batch job.
+        """
+        return {'dataset': self.name}
+
+    def get_job_prefix(self) -> str:
+        """
+        Prefix job names.
+        """
+        return f'{self.name}: '
+
+    def make_ped_file(self) -> Path:
+        """
+        Create a PED file for all samples
+        """
+        datas = []
+        for sample in self.get_samples():
+            if sample.pedigree:
+                datas.append(sample.pedigree.get_ped_dict())
+        df = pd.DataFrame(datas)
+
+        ped_path = self.get_tmp_bucket() / f'{self.name}.ped'
+        with ped_path.open('w') as fp:
+            df.to_csv(fp, sep='\t', index=False)
+
+        return ped_path
+
+
+class Sex(Enum):
+    """
+    Sex as in PED format
+    """
+
+    UNKNOWN = 0
+    MALE = 1
+    FEMALE = 2
+
+    @staticmethod
+    def parse(sex: str | None) -> 'Sex':
+        """
+        Parse a string into a Sex object.
+        """
+        if sex:
+            if sex.lower() in ('m', 'male', '1'):
+                return Sex.MALE
+            if sex.lower() in ('f', 'female', '2'):
+                return Sex.FEMALE
+        return Sex.UNKNOWN
```

Comment:
I think UNKNOWN should be a specific case handled within the conditional, and unrecognisable values should be an exception:

```suggestion
            if sex.lower() in ('u', 'unknown', '0'):
                return Sex.UNKNOWN
            raise ValueError(f'Unrecognised sex value {sex}')
        return Sex.UNKNOWN
```

Reply:
Agree!